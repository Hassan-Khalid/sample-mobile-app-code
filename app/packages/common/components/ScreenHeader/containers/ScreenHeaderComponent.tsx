import React, {  useMemo } from 'react';
import { Image, TouchableOpacity, View } from 'react-native';
import { Header } from '@rneui/themed';
import {  ScreenHeader } from 'common/declarations/types';
import styles from '../styles/ScreenHeaderComponent.styles';
import Text from "../../Text/Text"
import { COMMON_LOGOS } from 'common/assets/Images';
import { NavigationService } from '../../../services';
import useScreenHeader from 'common/components/ScreenHeader/hooks/useScreenHeader';
import CustomInput from 'common/components/InputText/InputText';
import ComponentBuilder from 'common/builder/ComponentBuilder';



const ScreenHeaderComponent = (props: ScreenHeader) => {
const {setSearchState,renderSearchBar,handleSearchTextChange,searchInputValue,toggleHandler} = useScreenHeader(props)

  const LeftComponent = ()=> {
    if(renderSearchBar){
      return (
        <View style={styles.leftContainer}>
         <Image source={COMMON_LOGOS.IC_SEARCH} resizeMode={'contain'} style={styles.icon} />
        </View>
      )
    }
    else {
      return (
        <View style={styles.leftContainer}>
          <TouchableOpacity onPress={() => NavigationService.openDrawer()}>
            <Image source={COMMON_LOGOS.IC_LIST} resizeMode={'contain'} style={styles.icon} />
          </TouchableOpacity>
        </View>
      )
    }
  }
  const CenterComponent = ()=>{
    if(renderSearchBar){
      return (
        <View style={styles.bodyContainer}>
        <SearchBarView/>
        </View>
      )
    }
    return (
      <View style={styles.bodyContainer}>
        <Text style={styles.headingText}>{props.title}</Text>
      </View>
    )
  }
  const RightComponent = ()=> {
    const rightLogo = renderSearchBar ? COMMON_LOGOS.IC_CROSS : COMMON_LOGOS.IC_SEARCH
    
      return (
        <View style={styles.rightContainer}>
          <TouchableOpacity onPress={toggleHandler}>
            <Image source={rightLogo} resizeMode={'contain'} style={renderSearchBar ? styles.crossIcon : styles.icon}/>
          </TouchableOpacity>
        </View>
      )
    
  }
  
  const SearchBarView = () =>{
    return(
        <CustomInput
         {...ComponentBuilder.createTestAttributes('Search_bar_work_order')}
          value={searchInputValue}
          placeholder={'Search here'}
          containerStyle={styles.searchInputContainer}
          inputContainer={styles.searchInput}
          onChangeText={handleSearchTextChange}
          blurOnSubmit={false}
      />
    )
  }
    return (
      <Header
        {...ComponentBuilder.createTestAttributes('header_component')}
        containerStyle={styles.container}
        placement={'left'}
        leftComponent={useMemo(()=> <LeftComponent />,[renderSearchBar,searchInputValue])}
        centerComponent={<CenterComponent />}
        rightComponent={useMemo(()=><RightComponent />,[renderSearchBar,searchInputValue])}
      />
    );
  // }
};

export default ScreenHeaderComponent;
