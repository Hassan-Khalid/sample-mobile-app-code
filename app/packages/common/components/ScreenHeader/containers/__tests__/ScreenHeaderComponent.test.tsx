import React from 'react';
import { render } from '@testing-library/react-native';
import ScreenHeaderComponent from '../ScreenHeaderComponent';

describe('ScreenHeaderComponent', () => {
  it('should render correctly with default props', () => {
    const wrapper = render(<ScreenHeaderComponent title="Test Title" />);
    
    // Check for the presence of header component
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
});
