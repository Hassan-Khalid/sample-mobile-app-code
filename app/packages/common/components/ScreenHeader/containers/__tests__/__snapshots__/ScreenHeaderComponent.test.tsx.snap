// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHeaderComponent should render correctly with default props 1`] = `
<View
  accessibilityLabel="header_component"
  style={
    {
      "alignItems": "center",
      "backgroundColor": {
        "dynamic": {
          "dark": "#1b1b1b",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  testID="header_component"
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "off",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
        }
      }
    >
      <View
        style={
          {
            "marginLeft": 10,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "testUri": "../../../packages/common/assets/Images/common/list.png",
              }
            }
            style={
              {
                "height": 22,
                "resizeMode": "contain",
                "width": 22,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flex": 3,
          "paddingHorizontal": 15,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "auto",
              "color": {
                "dynamic": {
                  "dark": "#172B4D",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#172B4D",
                },
              },
              "fontFamily": "Manrope-Regular",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Test Title
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "marginRight": 10,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "testUri": "../../../packages/common/assets/Images/common/search.png",
              }
            }
            style={
              {
                "height": 22,
                "resizeMode": "contain",
                "width": 22,
              }
            }
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
